src/hooks/useGetWalletNfts.ts

```diff
@@ -54,66 +52,84 @@ const useGetWalletNfts = () => {
+   */
+  const getTokenIdsByIdentifier = (identifier: Nft['identifier']) => {
+    return state.tokenIds[identifier] || []
+  }
 
-          const tokenIdPromises = []
+  /**
+   * Helper function to return all nfts in the config that are in the tokenIds
+   */
+  const getNftsInWallet = () => {
+    const identifiers = Object.keys(state.tokenIds)
+    return Nfts.filter((nft) => identifiers.includes(nft.identifier))
+  }
 
-          for (let i = 0; i < balanceOf; i++) {
-            tokenIdPromises.push(getTokenIdAndBunnyId(i))
+  useEffect(() => {
+    const fetchCollectibles = async () => {
+      // For each nft source get nft data
+      Object.keys(nftSources).forEach(async (nftSourceType) => {
+        const { address: addressObj } = nftSources[nftSourceType as NftType]
+        const address = getAddress(addressObj)
+        const contract = getErc721Contract(address)
+
+        const getTokenIdAndData = async (index: number) => {
+          try {
+            const tokenId = await contract.methods.tokenOfOwnerByIndex(account, index).call()
+            const walletNft = await getNftByTokenId(address, tokenId)
+            return [Number(tokenId), walletNft.identifier]
+          } catch (error) {
+            console.error('getTokenIdAndData', error)
+            return null
           }
+        }
 
-          const tokenIdsOwnedByWallet = await Promise.all(tokenIdPromises)
+        try {
+          const balanceOfResponse = await contract.methods.balanceOf(account).call()
+          const balanceOf = Number(balanceOfResponse)
 
-          const nfts: NftMap = tokenIdsOwnedByWallet.reduce((accum, association) => {
-            if (!association) {
-              return accum
+          if (balanceOf > 0) {
+            const nftDataFetchPromises = []
+
+            // For each index get the tokenId and data associated with it
+            for (let i = 0; i < balanceOf; i++) {
+              nftDataFetchPromises.push(getTokenIdAndData(i))
             }
 
-            const [bunnyId, tokenId, tokenUri] = association
+            const walletNftData = await Promise.all(nftDataFetchPromises)
 
-            return {
-              ...accum,
-              [bunnyId]: {
-                tokenUri,
-                tokenIds: accum[bunnyId] ? [...accum[bunnyId].tokenIds, tokenId] : [tokenId],
-              },
-            }
-          }, {})
+            // Format final values and add them to state
+            const nftTokenIdsByIdentifier: NftDataMap = walletNftData.reduce((accum, association) => {
+              if (!association) {
+                return accum
+              }
 
-          dispatch({ type: 'set_nfts', data: nfts })
-        } else {
-          // Reset it in case of wallet change
+              const [tokenId, identifier] = association as [number, Nft['identifier']]
+
+              return {
+                ...accum,
+                [identifier]: accum[identifier] ? [...accum[identifier].tokenIds, tokenId] : [tokenId],
```

Comment:
Not sure if the cough syrup is getting to me, but how come we expect the accumulated value for identifier to be an object, but return an array?

i.e shouldn't it be this
```typescript
[...accum[identifier], tokenId]
```
instead of this?
```typescript
[...accum[identifier].tokenIds, tokenId]
```

Reply:
Oh nice catch! It used to be that but I changed it and missed this. Thanks!